corehq/motech/dhis2/tasks.py

```diff
@@ -230,6 +228,21 @@ def get_previous_quarter(send_date: date) -> DateSpan:
     return DateSpan(startdate, enddate)
 
 
+def get_period(frequency: str, startdate: date) -> str:
+    if frequency == SEND_FREQUENCY_WEEKLY:
+        week_num = int(startdate.strftime('%W')) + 1
+        return startdate.strftime('%Y') + f'W{week_num}'
+
+    if frequency == SEND_FREQUENCY_MONTHLY:
+        return startdate.strftime('%Y%m')
+
+    if frequency == SEND_FREQUENCY_QUARTERLY:
+        quarter = (startdate.month - 1) // 3 + 1
```

Comment:
This calculates `quarter` correctly for all dates, although the previous (incorrect) calculation gave the right answer because it is only ever used with the first month in a quarter.